src/hotspot/cpu/riscv/jniFastGetField_riscv.cpp

```diff
@@ -175,9 +175,7 @@ address JNI_FastGetField::generate_fast_get_int_field0(BasicType type) {
     __ enter();
     ExternalAddress target(slow_case_addr);
     __ relocate(target.rspec(), [&] {
-      int32_t offset;
-      __ la(t0, target.target(), offset);
-      __ jalr(x1, t0, offset);
+      __ call(target.target());
```

Comment:
Should we revert this change after your last commit? As I think call is now not necessarily la + jalr.

Reply:
The addresses should come from JNI_FastGetField::generate_fast_get_XXX_field0.
So slow_case_addr is not really an ExternalAddress.
This means call will always do auipc + jalr as you know (intra code cache).

Reply:
Hmm... So I did a quick try on linux-aarch64 invoking `CodeCache::contains` on slow_case_addr and the result is false. Anything I missed?

Reply:
The JNI_FastGetField::generate_fast_get_XXX_field0 write the code in a CodeBuffer, this where slow_case_addr points to.

As I test with ReservedCodeCacheSize=2047M if we generated an li() relocation would fail. (there is usually around 120 MB between them)

Added your assert, it passes also.
```
     ExternalAddress target(slow_case_addr);
+    assert(CodeCache::contains(slow_case_addr), "Must be");
     __ relocate(target.rspec(), [&] {

```

I guess you are running on apple, the code concerning "static_fast_get_field_wrapper" smells.
Maybe you found a bug here.

Reply:
Try this: `make test TEST="runtime/jni/FastGetField/FastGetField.java"`
The assertion failed on both linux-aarch64 and linux-riscv64 platforms.

Reply:
As you turn off "fast JNI" with those option I think it's fine to use rt_call, which would do movptr in this case.
If you wanted fast JNI you should,'t turn it off.

But I can revert to la+jalr if you feel strongly about it?

Reply:
> But I can revert to la+jalr if you feel strongly about it?

Yeah, I would suggest revert to the original la+jalr for now and think more about it.

Reply:
Yes, let's finish this PR and make other wanted changes in other PR's :)